### PR TITLE
Add method to calculate perplexity of distribution

### PR DESCRIPTION
**Pull request by zuoxingdong**

### CODE REVIEW

**Comment by apaszke** [comment]
cc: @fritzo

File: torch/distributions/distribution.py [hunk 1]
```diff
@@ -163,6 +163,12 @@ def entropy(self):
             Tensor of shape batch_shape.
         """
         raise NotImplementedError
+        
+    def perplexity(self):
+        """
+        Returns perplexity of distribution
```
Comment (fritzo) [inline torch/distributions/distribution.py]: Could you match the `.entropy()` docs:
> Returns perplexity of distribution, batched over batch_shape.

**Comment by apaszke** [comment]
Actually isn't it the case that `perplexity = 2^entropy`?

**Comment by fritzo** [comment]
We measure entropy in nats rather than bits, so exp is correct.

**Review by apaszke — APPROVED** [review]

**Pull request merged**